DashAI/back/models/hugging_face/distilbert_transformer.py

```diff
@@ -0,0 +1,176 @@
+        input_column = train_dataset.inputs_columns[0]
+        output_column = train_dataset.outputs_columns[0]
+
+        tokenizer_func = self.get_tokenizer(input_column, output_column)
+        train_dataset = train_dataset.map(
+            tokenizer_func, batched=True, batch_size=len(train_dataset)
+        )
+        train_dataset.set_format(
+            "torch", columns=["input_ids", "attention_mask", "labels"]
+        )
+
+        # Arguments for fine-tuning
+        training_args = TrainingArguments(
+            output_dir="DashAI/back/user_models/temp_checkpoints_distilbert",
+            save_steps=1,
+            save_total_limit=1,
+            **self.training_args,
+        )
+
+        # The Trainer class is used for fine-tuning the model.
+        trainer = Trainer(
+            model=self.model,
+            args=training_args,
+            train_dataset=train_dataset,
+        )
+
+        trainer.train()
+        self.fitted = True
+        shutil.rmtree(
+            "DashAI/back/user_models/temp_checkpoints_distilbert", ignore_errors=True
+        )
+        return
+
+    def predict(self, dataset: DatasetDict):
+        """Predicting with the fine-tuned model
+
+        Parameters
+        ----------
+        dataset : DatasetDict
+            Datasetdict with training data
+
+        Returns
+        -------
+        Numpy Array
+            Numpy array with the probabilities for each class
+        """
+        if not self.fitted:
+            raise NotFittedError(
+                f"This {self.__class__.__name__} instance is not fitted yet. Call 'fit'"
+                " with appropriate arguments before using this "
+                "estimator."
+            )
+
+        test_dataset = dataset["test"]
+        input_column = test_dataset.inputs_columns[0]
+        output_column = test_dataset.outputs_columns[0]
+        tokenizer_func = self.get_tokenizer(input_column, output_column)
+        test_dataset = test_dataset.map(
+            tokenizer_func, batched=True, batch_size=len(test_dataset)
+        )
+        test_dataset.set_format(
+            "torch", columns=["input_ids", "attention_mask", "labels"]
+        )
+
+        probabilities = []
+
+        # Iterate over each batch in the dataset
+        for batch in test_dataset:
+            # Make sure that the tensors are in the correct device.
+            batch = {k: v.to(self.model.device) for k, v in batch.items()}
+
+            outputs = self.model(**batch)
+
+            # Takes the model probability using softmax
+            probs = outputs.logits.softmax(dim=-1)
+
+            probabilities.extend(probs.detach().cpu().numpy())
+        return np.array(probabilities)
+
+    def save(self, filename=None):
+        self.model.save_pretrained(filename)
+
+    @classmethod
+    def load(cls, filename):
+        model = DistilBertForSequenceClassification.from_pretrained(filename)
+        return cls(model=model)
```

Comment:
This method are deprecated and they are not required anymore.
```suggestion
```

Reply:
My problem with eliminating this method is that you would not be able to test as well that the save method is working.